R/utilities-simulation-results.R

```diff
@@ -0,0 +1,129 @@
+  validateIsOfType(simulationResults, "SimulationResults")
+
+  # If multiple individuals were simulated, 'getPopulationResultsValues' should be called
+  if (simulationResults$count > 1) {
+    stop(messages$errorGetResultValuesPopulationSimulation)
+  }
+
+  output <- getPopulationResultsValues(simulationResults, quantitiesOrPaths)
+
+  return(output)
+}
+
+#' Get simulated values of molecules or parameters.
+#'
+#' Return values for given entities generated in a population simulation.
+#'
+#' The function receives an object of simulation results generated by running the simulation
+#' and returns time-values profiles for the chosen quantities. Results of a simulation
+#' of a single individual is treated as a population simulation with only one individual.
+#'
+#' @param simulationResults Object of type \code{SimulationResults} produced by calling \code{runSimulation}
+#' on a \code{Simulation} object.
+#' @param quantitiesOrPaths Quantity instances (element or vector) (typically retrieved using
+#' \code{getAllQuantitiesMatching}) or quantity path (element or vector of strings) for which the results are to be returned.
+#' When providing the paths, only absolute full paths are supported (i.e., no matching with '*' possible)
+#' @param individualIds \code{numeric} IDs of individiuals for which the results should be extracted.
+#' By default, all individuals from the results are considered. If the individual with the provided ID is not found, the ID is ignored
+#'
+#' @return A list of results, each entry having the name of a quantity path and consisting of the element \code{x} being the time values (in minutes)
+#' and the element \code{y} a \code{NxM} matrix of simulated values, with \code{N} the number of entries (equals the number of output time steps)
+#' and \code{M} the number of individuals. If no individual contains the results for a given path, the value of the element is \code{NULL}
+#'
+#' @seealso \code{\link{getResultsValues}}
+#'
+#' @export
+#'
+#' @examples
+#' simPath <- system.file("extdata", "simple.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simPath)
+#' simResults <- runSimulation(sim)
+#'
+#' paths <- "Organism|Volume"
+#'
+#' resultsValues <- getPopulationResultsValues(simResults, paths, individualIds = 0)
+#' plot(resultsValues[[paths]])
+getPopulationResultsValues <- function(simulationResults, quantitiesOrPaths, individualIds = NULL) {
+  validateIsOfType(simulationResults, "SimulationResults")
+  quantitiesOrPaths <- c(quantitiesOrPaths)
+  validateIsOfType(quantitiesOrPaths, c("Quantity", "character"))
+  validateIsNumeric(individualIds, nullAllowed = TRUE)
+
+  # If quantities are passed, get their paths.
+  paths <- quantitiesOrPaths
+  if (isOfType(quantitiesOrPaths, "Quantity")) {
+    paths <- unlist(lapply(quantitiesOrPaths, function(x) x$path))
+    # Remove the name of the simulation
+    splitPaths <- strsplit(paths, "|", fixed = TRUE, perl = FALSE, useBytes = FALSE)
```

Comment:
Don't really like this one, but I didn't come up with a better solution right now.

Reply:
If you go to entity class, we have
```
path = function(value) {
      private$wrapExtensionMethod("OSPSuite.Core.Domain.EntityExtensions", "EntityPath", "path", value)
    }
```

You can define another method using the "ConsolidatedPath" method
It removes the first entry effectively of the path for you

Something like
```
consolidatePath = function(value) {
      private$wrapExtensionMethod("OSPSuite.Core.Domain.EntityExtensions", "ConsolidatedPath", "consolidatePath", value)
    }
```

Don't forget to add the corresponding @field document and explain that this removes the simulation path. This should normally never be used outside of our helper methods